### ISSUE TITLE
Incorrect file name returned

### ISSUE THREAD

**Issue by GoogleCodeExporter** [bdshadow/gwtupload#16]
```
What steps will reproduce the problem?
1. Upload a file
2. onFinish(), call IUploader.getFileName()

 - What is the expected output? What do you see instead?
I would expect the file name of the file being uploaded. In some browsers,
(Firefox 3.5 on Linux) I get only the base file name. However, other
browsers (IE 8 on Windows XP), I get "C:\fakepath\{FILENAME}".

 - What version of the product are you using? On what operating system?
It's reproducible on all the tested versions (0.5.1 and 0.5.3).
The client/server OS is not important, it's a browser dependent issue.

 - Please provide any additional information below.
There are no points getting the local path of the file. This function
should always return the basename of the file from the client. If this
can't be fixed for some reasons, it would be very convenient to get a new
function returning the basename.

```

Original issue reported on code.google.com by `iscy%inv...@gtempaccount.com` on 20 Oct 2009 at 7:56

**Comment by GoogleCodeExporter** [comment]
```
Hi,
Unfortunately there is nothing to do with this issue.  Gwtupload just uses 
FileUpload
widget which asks the browser for the file selected. Each browser has its own
behavior and could return a different value because of security restrictions. 
Its
known that every browsers return at least the basename of the file, but the 
dirname
is only reported by some of them.

Thanks for using gwtupload.
Manolo
```

Original comment by `manuel.carrasco.m` on 23 Oct 2009 at 8:32
- Changed state: **Invalid**

**Comment by GoogleCodeExporter** [comment]
```
What about adding a method to return only the base file name?
```

Original comment by `iscy%inv...@gtempaccount.com` on 23 Oct 2009 at 12:33

**Comment by GoogleCodeExporter** [comment]
```
There is already a method to do that in the client library to do that, but you 
have
to call it:
IUploader.Utils.basename(uploader.getFileName())

Maybe a good idea to add it to the uploader class.

Regards
Manolo
```

Original comment by `manuel.carrasco.m` on 23 Oct 2009 at 12:43

**Comment by GoogleCodeExporter** [comment]
```
Yes, we figured that one out by looking at the code in the Uploader. That's 
what is
used to display the status while the upload is happening. It's just not as 
obvious as
the rest of the API.
```

Original comment by `iscy%inv...@gtempaccount.com` on 23 Oct 2009 at 12:48

**Comment by GoogleCodeExporter** [comment]
```
You are right, this feature could be useful in the uploadar, so I'll put this 
feature
in the next release. I'll close this issue when it is in the repository.

Thanks for the suggestion
Manolo
```

Original comment by `manuel.carrasco.m` on 23 Oct 2009 at 8:08
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Thank you,
```

Original comment by `iscy%inv...@gtempaccount.com` on 23 Oct 2009 at 8:12

**Comment by GoogleCodeExporter** [comment]
```
I've committed the change to the repository, so wait till a new version is 
uploaded
or checkout from trunk.


Manolo
```

Original comment by `manuel.carrasco.m` on 30 Oct 2009 at 1:26
- Changed state: **FixedinSVN**

**Comment by GoogleCodeExporter** [comment]
```
released version 0.5.5 which includes this feature
```

Original comment by `manuel.carrasco.m` on 12 Nov 2009 at 10:43
- Changed state: **Fixed**
